lib/smart_proxy_ansible/runner/ansible_runner.rb

```diff
@@ -303,6 +304,29 @@ def rebuild_secrets(inventory, input)
 
         inventory
       end
+
+      def prune_known_hosts_on_first_execution(hosts)
+        hosts.each do |value|
+          action_input = value[:input][:action_input]
+          next unless action_input.fetch(:first_execution, false)
+
+          hostname = action_input.fetch(:name)
+          host_ip = action_input.fetch(:hostname)
+          ansible_host = fetch_ansible_host_attribute(action_input, hostname, "ansible_host")
+          ssh_port = action_input.fetch(:ssh_port, 22)
+          ansible_port = fetch_ansible_host_attribute(action_input, hostname, ansible_port) || 22
```

Comment:
If I'm reading it right the way it is now it would always default to `22`
```suggestion
          ansible_port = fetch_ansible_host_attribute(action_input, hostname, "ansible_port") || 22
```